great_expectations_provider/operators/great_expectations.py

```diff
@@ -44,121 +78,404 @@ class GreatExpectationsOperator(BaseOperator):
+        conn_id: Optional[str] = None,
+        execution_engine: Optional[str] = None,
+        expectation_suite_name: Optional[str] = None,
+        data_asset_name: Optional[str] = None,
         data_context_root_dir: Optional[Union[str, bytes, os.PathLike]] = None,
         data_context_config: Optional[DataContextConfig] = None,
+        dataframe_to_validate: Optional[
+            DataFrame
+        ] = None,  # should we allow a Spark DataFrame as well?
+        query_to_validate: Optional[str] = None,
         checkpoint_name: Optional[str] = None,
         checkpoint_config: Optional[CheckpointConfig] = None,
-        checkpoint_kwargs: Optional[Dict] = None,
-        fail_task_on_validation_failure: Optional[bool] = True,
+        checkpoint_kwargs: Optional[Dict[str, Any]] = None,
         validation_failure_callback: Optional[
             Callable[[CheckpointResult], None]
         ] = None,
+        fail_task_on_validation_failure: bool = True,
         return_json_dict: bool = False,
-        **kwargs
-    ):
-        super().__init__(**kwargs)
+        use_open_lineage: bool = True,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
 
+        self.data_asset_name: Optional[str] = data_asset_name
         self.run_name: Optional[str] = run_name
+        self.conn_id: Optional[str] = conn_id
+        self.execution_engine: Optional[str] = (
+            execution_engine if execution_engine else "PandasExecutionEngine"
+        )
+        self.expectation_suite_name: Optional[str] = expectation_suite_name
         self.data_context_root_dir: Optional[
-            Union[str, bytes, os.PathLike]
+            Union[str, bytes, os.PathLike[Any]]
         ] = data_context_root_dir
-        self.data_context_config: DataContextConfig = data_context_config
+        self.data_context_config: Optional[DataContextConfig] = data_context_config
+        self.dataframe_to_validate: Optional[DataFrame] = dataframe_to_validate
+        self.query_to_validate: Optional[str] = query_to_validate
         self.checkpoint_name: Optional[str] = checkpoint_name
-        self.checkpoint_config: Optional[CheckpointConfig] = checkpoint_config or {}
-        self.checkpoint_kwargs: Optional[dict] = checkpoint_kwargs
+        self.checkpoint_config: Union[CheckpointConfig, Dict[Any, Any]] = (
+            checkpoint_config if checkpoint_config else {}
+        )
+        self.checkpoint_kwargs: Optional[Dict[str, Any]] = checkpoint_kwargs
         self.fail_task_on_validation_failure: Optional[
             bool
         ] = fail_task_on_validation_failure
         self.validation_failure_callback: Optional[
             Callable[[CheckpointResult], None]
         ] = validation_failure_callback
         self.return_json_dict: bool = return_json_dict
+        self.use_open_lineage = use_open_lineage
+        self.is_dataframe = True if self.dataframe_to_validate is not None else False
```

Comment:
Would it be possible to avoid `is_dataframe` flag by proper logic on the value of the `dataframe_to_validate` argument?  Thanks.